goald/quality/pragmatic/model/task.py

```diff
@@ -49,5 +47,34 @@ def myProvidedQuality(self, metric, contextSet):
                             myQuality = self.providedQualityLevels[metric][current]
                     elif(myQuality < self.providedQualityLevels[metric][current]):
                         myQuality = self.providedQualityLevels[metric][current]
-    
+
         return myQuality
+
+    def abidesByInterpretation(self, interp, current):
+        feasible = True
+        if interp is None:
+            return True
+
+        for qc in interp.getQualityConstraints(current):
+            try:
+                if not qc.abidesByQC(self.myProvidedQuality(qc.getMetric(), current), qc.getMetric()):
+                    feasible = False
+            except:
+                print("MetricNotFoundException")
+        if interp.getQualityConstraints(None) != None:
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [comparison to None should be 'if cond is not None:'](https://app.codacy.com/gh/lesunb/p2psimpy/pullRequest?prid=4861978)